src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -266,30 +261,71 @@ public function execute()
             return [];
         }
 
-        $this->queryBuilder = $this->createFullQueryBuilder(
+        $this->queryBuilder = $this->createNonGroupQueryBuilder(
             $this->em->createQueryBuilder()->from($this->entityName, $this->encodeAlias($this->entityName))
         );
 
         // now select all data
-        $this->assignJoins($this->queryBuilder);
+        $this->assignJoins($this->queryBuilder, $this->getNonGroupJoins());
 
         // use ids previously selected ids for query
         $select = $this->idField->getSelect();
         $this->queryBuilder->where($select . ' IN (:ids)')->setParameter('ids', $ids);
 
         $this->assignParameters($this->queryBuilder);
 
-        return $this->queryBuilder->getQuery()->getArrayResult();
+        $nonGroupResult = $this->queryBuilder->getQuery()->getArrayResult();
+
+        if (!$this->hasGroupingFieldDescriptor()) {
+            return $nonGroupResult;
+        }
+
+        $this->queryBuilder = $this->createGroupQueryBuilder(
+            $this->em->createQueryBuilder()->from($this->entityName, $this->encodeAlias($this->entityName))
+        );
+
+        // now select all data
+        $this->assignJoins($this->queryBuilder, $this->getGroupJoins());
+
+        // use ids previously selected ids for query
+        $select = $this->idField->getSelect();
+        $this->queryBuilder->where($select . ' IN (:ids)')->setParameter('ids', $ids);
+
+        $this->assignParameters($this->queryBuilder);
+
+        $groupResult = $this->queryBuilder->getQuery()->getArrayResult();
+
+        $result = [];
+        for ($i = 0; $i < \count($nonGroupResult); ++$i) {
+            $result[] = \array_merge($nonGroupResult[$i], $groupResult[$i]);
```

Comment:
This look risky as we are not checking if `groupResult` matches to the same id as the `nonGroupResult`.

Reply:
@alexander-schranz your right - have changed it to indexed query! should be safe now